### PR TITLE
UefiPayloadPkg: Add setup menu password driver

### PR DESCRIPTION
**Pull request by mkopec**
Depends on https://github.com/Dasharo/DasharoModulePkg/pull/5

Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>

### CODE REVIEW

File: UefiPayloadPkg/UefiPayloadPkg.dsc [hunk 1]
```diff
@@ -94,6 +94,7 @@
   DEFINE LOAD_OPTION_ROMS               = TRUE
   DEFINE SECURE_BOOT_DEFAULT_ENABLE     = TRUE
   DEFINE DASHARO_SYSTEM_FEATURES_ENABLE = FALSE
+  DEFINE SETUP_PASSWORD_ENABLE          = FALSE
```
Comment (miczyg1) [inline UefiPayloadPkg/UefiPayloadPkg.dsc]: Remove it, we actually want to have a password compiled by default, as it is a standard feature.

File: UefiPayloadPkg/UefiPayloadPkgIa32X64.dsc [hunk 2]
```diff
@@ -94,6 +94,7 @@
   DEFINE LOAD_OPTION_ROMS               = TRUE
   DEFINE SECURE_BOOT_DEFAULT_ENABLE     = TRUE
   DEFINE DASHARO_SYSTEM_FEATURES_ENABLE = FALSE
+  DEFINE SETUP_PASSWORD_ENABLE          = FALSE
```
Comment (miczyg1) [inline UefiPayloadPkg/UefiPayloadPkgIa32X64.dsc]: Changes in this file are no longer needed. We do symlink in the PR with Secure Boot

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: UefiPayloadPkg/UefiPayloadPkg.dsc (hunk 1 above)
Comment (mkopec) [reply UefiPayloadPkg/UefiPayloadPkg.dsc]: Ok, removed the option entirely

File: UefiPayloadPkg/UefiPayloadPkgIa32X64.dsc (hunk 2 above)
Comment (mkopec) [reply UefiPayloadPkg/UefiPayloadPkgIa32X64.dsc]: Ack

Comment (mkopec) [reply UefiPayloadPkg/UefiPayloadPkgIa32X64.dsc]: Reverted the changes to this file. Of course now it won't work until the secure boot patches get merged.

**Review by miczyg1 — APPROVED** [review]

File: UefiPayloadPkg/UefiPayloadPkg.dsc [hunk 3]
```diff
@@ -570,6 +570,12 @@
   SecurityPkg/VariableAuthenticated/SecureBootDefaultKeysDxe/SecureBootDefaultKeysDxe.inf
 !endif
 
+!include DasharoModulePkg/DasharoModuleComponents.dsc.inc
+!include DasharoModulePkg/Include/UserAuthFeature.dsc
```
Comment (miczyg1) [inline UefiPayloadPkg/UefiPayloadPkg.dsc:574]: I don't like this actually.

UserAuthFeature.dsc contains `[LibraryClasses]` and `[Components.X64]` which can't be included here (one should keep ordering of the LibraryClasses and Components sections and not mix them).
Same with `DasharoModuleComponents.dsc.inc`, it includes libraries and modules that are duplicated in the UefiPayloadPkg.dsc. 

Include the Dasharo password modules directly here without including DSC.

File: UefiPayloadPkg/UefiPayloadPkgIa32.dsc [hunk 4]
```diff
@@ -505,6 +505,12 @@
   OvmfPkg/EnrollDefaultKeys/EnrollDefaultKeys.inf
 !endif
 
+!include DasharoModulePkg/DasharoModuleComponents.dsc.inc
+!include DasharoModulePkg/Include/UserAuthFeature.dsc
```
Comment (miczyg1) [inline UefiPayloadPkg/UefiPayloadPkgIa32.dsc:509]: One of the DSC fiel has component.x64 definition. How it is supposed to work in IA32? Remove all modifications from this file as we do not use 32bit payload anyways.

**Review by miczyg1 — CHANGES_REQUESTED** [review]

**Pull request merged**